pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -129,6 +132,12 @@ type OFBridge struct {
 	connCh chan struct{}
 	// connected is an internal channel to notify if connected to the OFSwitch or not. It is used only in Connect method.
 	connected chan bool
+	// pktConsumers is a map from PacketIn reason to the channel that is used to publish the PacketIn message.
+	pktConsumers map[uint8]chan *ofctrl.PacketIn
```

Comment:
doesn't this map need to be made thread-safe by protecting it with a mutex?

Reply:
It seems to me that there can be concurrent access between `SubscribePacketIn` and `PacketRcvd`, they both access the map and I expect that they are being called by different goroutines. Unless you want to make the assumption that all subscriptions (calls to `SubscribePacketIn`) are made at initialization time, before there can be any callback to `PacketRcvd`.

Reply:
@antoninbas Yes, I assume `SubscribePacketIn` should be callled before PacketIn message are received. We are using action "SendToController" to require OVS to send Packet to Antrea, and I expect this action is called after the OpenFlow client is ready to receive packets.

Actually there is no harm if the map is changed to use sync.Map, I could modify it if you prefer.

Reply:
In case of an agent restart (agent, not ovs-vswitchd), is it possible for `PacketRcvd` to be called even before the new call(s) to `SubscribePacketIn`, because of pre-existing flows, in which case we could have a race?

Reply:
Make sense, then I will change to use sync.Map.